customerio-reactnative.podspec

```diff
@@ -19,6 +19,15 @@ Pod::Spec.new do |s|
 
   # Syntax of native iOS pods allows for automatically upgrading to the latest major version of the iOS SDK. 
   # Reference: https://guides.cocoapods.org/syntax/podfile.html#pod
-  s.dependency "CustomerIO/Tracking", '~> 2'
-  s.dependency "CustomerIO/MessagingInApp", '~> 2'
+  s.dependency "CustomerIO/Tracking", package["cioNativeiOSSdkVersion"]
+  s.dependency "CustomerIO/MessagingInApp", package["cioNativeiOSSdkVersion"]
+
+  s.subspec 'apn' do |ss|
+    ss.dependency "CustomerIO/MessagingPushAPN", package["cioNativeiOSSdkVersion"]
+  end
+
+  s.subspec 'fcm' do |ss|
+    ss.dependency "CustomerIO/MessagingPushFCM", package["cioNativeiOSSdkVersion"]
+    ss.dependency "FirebaseMessaging"
+  end
```

Comment:
Added subspecs which is a cocoapods feature allowing you to create variants of SDKs. 

This allows us to conveniently create a APN and separate FCM cocoapod in 1 `podspec` file.